Update introduction. (#1075)
quickwit-inc/quickwit: docs/introduction.md

---
title: Introduction
slug: /
sidebar_position: 1
---

Quickwit is a distributed search engine designed from the ground up to offer cost-efficiency and high reliability on large data sets.

> Quickwit is particularly well-suited for dealing with large, immutable datasets and a low average QPS$^1$. Its benefits are most apparent in a multi-tenancy or multi-index setting.

Reviewer: ```suggestion
Quickwit is particularly well-suited for dealing with large, immutable datasets and relatively low average QPS$^1$. Its benefits are most apparent in multi-tenancy or multi-index settings.
```

Reviewer 2: The ^1 thing is broken.

Common use cases for Quickwit include:

> - Searching through logs, from a small amount to TB of data.

Reviewer: ```suggestion
- Searching through logs, from small amounts of data to terabytes.
```

- Adding full-text search capabilities to [OLAP databases such as ClickHouse](link to blog post).

> - Searching through backups sitting on your cloud storage by adding Quickwit index files on your same storage.

Reviewer: This needs to be reworded but I have a hard time coming up with something compelling.

Reviewer 2: Broken link

Author: Link fixed.

# Key features of Quickwit

> Quickwit allows true decoupled compute and storage and we designed it to search straight from deep storage (cloud storage, HDFS, local file system, ...) in a stateless manner. Here is a non-exhaustive list of Quickwit’s key features:

Reviewer: ```suggestion
Quickwit is designed to search straight from object storage allowing true decoupled compute and storage. Here is a non-exhaustive list of Quickwit’s key features:
```

> - **Scalable distributed search:** Host an arbitrary amount of indices on S3 and query them with a pool of search servers and with sub-second latency.

Reviewer: ```suggestion
- **Scalable distributed search:** Host an arbitrary number of indexes on Amazon S3 and answer search queries in less than a second with a small pool of stateless search instances.
```

> - **Stream indexing:** Plug in your distributed event streaming platform and ingest TB of data. As of today, Quickwit supports Kafka natively.

Reviewer: ```suggestion
- **Stream indexing:** Ingest TB of data from your favorite distributed event streaming service. As of today, Quickwit supports Apache Kafka natively. The next releases will bring support for more platforms.
```

> - **Fault-tolerant architecture that won't lose data:** Quickwit ensures **exactly-once** semantics for indexing and your data can be safely stored on highly reliable deep storage like S3.

Reviewer: ```suggestion
- **Fault-tolerant architecture that won't lose data:** Quickwit achieves **exactly-once** processing for indexing and safely stores your data on highly reliable object storage services such as Amazon S3.
```

> - **Cloud-native, easy to operate:** Thanks to true decoupled compute and storage, search instances are stateless, you can add or remove search instances in seconds.

Reviewer: ```suggestion
- **Cloud-native, easy to operate:** Thanks to true decoupled compute and storage, search instances are stateless, add or remove search nodes within seconds.
```

- **Sub-second full-text search on cloud / distributed storage:** Quickwit Search re-designed indexing and index data structure to open it in less than 60ms on Amazon S3**.**
- **Time-based sharding:** Quickwit shards data by time when enabled. And you can use a second dimension to shard data thanks to our [tags feature](../design/querying.md). Time-based queries only access splits (a data piece of the index) that match the time range of the query which leads to significant performance improvements.
- **Painless multi-tenant search:** Create indexes for each tenant without hurting query performance. Or group tenants into one index and use tagging to prune irrelevant splits for your tenant query to improve significantly performance.

# When to use Quickwit

Quickwit should be a good match if your use case has some of the following characteristics:

- Your documents are immutable.
- You are targeting query latencies of 100ms to a few seconds.

> - You have a low average QPS$^1$, typically < 10 QPS on average over the month. This is the case for most search use cases as long as search is not public: enterprise search, log search, email search, security search, ...

Reviewer 2: broken footnote link (see above.)
